packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -44,39 +46,56 @@ export function mergeConfigs(...configObjects) {
 }
 
 /**
- * Extracts keys starting with a particular namespace from a flattened config
- * object, removing the namespace in the process.
+ * Extracts keys starting with a particular namespace from dataset ('data-*')
+ * object, removing the namespace in the process, normalising all values
  *
  * @internal
- * @param {{ [key: string]: unknown }} configObject - The object to extract key-value pairs from.
- * @param {string} namespace - The namespace to filter keys with.
- * @returns {{ [key: string]: unknown }} Flattened object with dot-separated key namespace removed
+ * @param {{ schema: Schema }} Component - Component class
+ * @param {DOMStringMap} dataset - The object to extract key-value pairs from
+ * @param {string} namespace - The namespace to filter keys with
+ * @returns {ObjectNested | undefined} Nested object with dot-separated key namespace removed
  */
-export function extractConfigByNamespace(configObject, namespace) {
-  /** @type {{ [key: string]: unknown }} */
-  const newObject = {}
+export function extractConfigByNamespace(Component, dataset, namespace) {
+  const property = Component.schema.properties[namespace]
+
+  // Only extract configs for object schema properties
+  if (property?.type !== 'object') {
+    return
```

Comment:
Is a blocker?

It makes sure missing configs are caught during development should the schema change:

```js
// Returns empty `{}` object by default
extractConfigByNamespace(Accordion, dataset, 'i18n')

// Returns `undefined` by default
extractConfigByNamespace(Accordion, dataset, 'abcd')
```

https://github.com/alphagov/govuk-frontend/blob/2c822ac4dc23fbdfb8d6370b0897c740ab51b90d/packages/govuk-frontend/src/govuk/common/index.jsdom.test.mjs#L163-L164

Reply:
Not massively blocking, was just trying to avoid running the same `if` twice in a row. Might be a signal that we'd want to group `extractConfigByNamespace` with `normaliseDataset` and not export it given that's all we use it for. But yeah, not blocking, just a suggestion 😊